### PR TITLE
Fix file creation in `quoted.QuoteCompiler`

### PR DESCRIPTION
**Pull request by michelou**
`"<quote>"` is not a legal file name on Windows.
We suggest to use `"_quote_"` instead (similar to path `"_site"` in doc-tool, see [`Site.scala`](https://github.com/lampepfl/dotty/blob/master/doc-tool/src/dotty/tools/dottydoc/staticsite/Site.scala#L130)).

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/quoted/QuoteCompiler.scala [hunk 1]
```diff
@@ -67,7 +67,7 @@ class QuoteCompiler extends Compiler {
       */
     private def inClass(expr: Expr[_])(implicit ctx: Context): Tree = {
       val pos = Position(0)
-      val assocFile = new PlainFile(Path("<quote>"))
+      val assocFile = new PlainFile(Path("_quote_"))
```
[the remark this answers is not in the document]
Comment (michelou) [reply compiler/src/dotty/tools/dotc/quoted/QuoteCompiler.scala]: Good point. I forget to mention the `VirtualFile` option.

[the remark this answers is not in the document]
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/quoted/QuoteCompiler.scala]: Yes,  a virtual file would be better.

File: compiler/src/dotty/tools/dotc/quoted/QuoteCompiler.scala [hunk 2]
```diff
@@ -67,7 +67,7 @@ class QuoteCompiler extends Compiler {
       */
     private def inClass(expr: Expr[_])(implicit ctx: Context): Tree = {
       val pos = Position(0)
-      val assocFile = new PlainFile(Path("<quote>"))
+      val assocFile = new VirtualFile(Path("<quote>").path)
```
[the remark this answers is not in the document]
Comment (michelou) [reply compiler/src/dotty/tools/dotc/quoted/QuoteCompiler.scala]: You're right! Update follows.

**Comment by nicolasstucki** [comment]
You still need to sign the [Scala CLA](https://www.lightbend.com/contribute/cla/scala)

**Pull request merged**